ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/helpers/MiscHelpers.java

```diff
@@ -114,15 +114,27 @@ public int computeProposerIndex(
     }
   }
 
-  public UInt64 computeEpochAtSlot(UInt64 slot) {
+  public UInt64 computeEpochAtSlot(final UInt64 slot) {
     return slot.dividedBy(specConfig.getSlotsPerEpoch());
   }
 
-  public UInt64 computeStartSlotAtEpoch(UInt64 epoch) {
+  public UInt64 computeStartSlotAtEpoch(final UInt64 epoch) {
     return epoch.times(specConfig.getSlotsPerEpoch());
   }
 
-  public UInt64 computeTimeAtSlot(BeaconState state, UInt64 slot) {
+  public UInt64 computeEndSlotAtEpoch(final UInt64 epoch) {
+    return computeStartSlotAtEpoch(epoch).plus(specConfig.getSlotsPerEpoch() - 1);
+  }
+
+  public UInt64 computeSlotAtTime(final UInt64 genesisTime, final UInt64 currentTime) {
+    return currentTime.minusMinZero(genesisTime).dividedBy(specConfig.getSecondsPerSlot());
+  }
+
+  public UInt64 computeTimeAtSlot(final UInt64 genesisTime, final UInt64 slot) {
+    return genesisTime.plus(slot.times(specConfig.getSecondsPerSlot()));
+  }
+
+  public UInt64 computeTimeAtSlot(final BeaconState state, final UInt64 slot) {
```

Comment:
test... they're simple in this case but we really do need basic tests so that when we refactor we can find issues.